src/domain_bridge/domain_bridge.cpp

```diff
@@ -412,9 +421,41 @@ class DomainBridgeImpl
 
         this->bridged_topics_[topic_bridge] = {publisher, subscription};
       };
-    wait_for_graph_events_.set_delay(topic_options.delay());
-    wait_for_graph_events_.register_on_publisher_qos_ready_callback(
-      topic, from_domain_node, create_bridge);
+    auto opt_case = static_cast<int>(topic_options.wait_for_subscription()) * 2 +
+      static_cast<int>(topic_options.wait_for_publisher());
+    switch (opt_case) {
+      case 0:  // DON'T WAIT
+        {
+          domain_bridge::QosMatchInfo qos_match;  // with default qos
+          create_bridge(qos_match);
+        }
+        break;
+      case 1:  // WAIT FOR ONLY PUBLISHER, this is the default
+        wait_for_graph_events_.set_delay(topic_options.delay());
+        wait_for_graph_events_.register_on_publisher_qos_ready_callback(
+          topic, from_domain_node, create_bridge);
+        break;
+      case 2:  // WAIT FOR ONLY SUBSCRIPTION
+        wait_for_graph_events_.set_delay(topic_options.delay());
+        wait_for_graph_events_.register_on_subscription_ready_callback(
+          topic, from_domain_node, [create_bridge]() {
+            domain_bridge::QosMatchInfo qos_match;  // with default qos
```

Comment:
Are there downsides to matching the subscription QoS?

Reply:
mmm, I had some doubts about that, but after thinking again I think there aren't

Reply:
Done in https://github.com/ros2/domain_bridge/pull/52/commits/ec986f36d7ce6de65962a51af83ef8e9b1db8e04.